celery/result.py

```diff
@@ -853,7 +855,11 @@ def __bool__(self):
 
     def __eq__(self, other):
         if isinstance(other, GroupResult):
-            return other.id == self.id and other.results == self.results
+            return (
```

Comment:
Perhaps add two simple test cases: one for `__eq__` and one for `as_tuple`. What do you think?